$MirOS$

	revision 1.107
	date: 2006/05/06 22:41:04;  author: jbo_85;  state: Exp;  lines: +1 -3
	Finally fixed the most annoying bug ever. ZSNES will no longer
	jump back into the GUI randomly if you press F2 / F4 or some
	other buttons repeatedly. Now you will never bother me again you
	piece of shit. Special thanks to Dan for the clues on how to
	recreate it.

--- src/linux/sdlintrf.asm.orig	Fri Jan 14 13:10:56 2005
+++ src/linux/sdlintrf.asm	Mon Nov  3 17:13:57 2008
@@ -534,9 +534,7 @@ NEWSYM Get_Key
     ; for extended keys, return a 0, then the extended key afterwards
     xor eax,eax
 .nokey
-    pushad
-    call JoyRead
-    popad
+;    call JoyRead
     mov al,[CurKeyReadPos]
     cmp al,[CurKeyPos]
     je .nokey
@@ -1219,24 +1217,20 @@ NEWSYM StartSound
     call JoyRead
     ret
 
-
 NEWSYM SoundProcess     ; This function is called ~60 times/s at full speed
-   pushad
     cmp byte[soundon],0
     je .nosound
     cmp byte[DSPDisable],1
     je .nosound
-    mov eax,256         ; Size
-    mov dword[BufferSizeB],eax
-    add eax,eax
-    mov dword[BufferSizeW],eax
+    mov dword[BufferSizeB],256
+    mov dword[BufferSizeW],512
     pushad
     call ProcessSoundBuffer
     popad
+
     ; DSPBuffer should contain the processed buffer in the specified size
     ; You will have to convert/clip it to 16-bit for actual sound process
 .nosound    
-    popad
     ret
 
 NEWSYM delay
